keras_core/layers/preprocessing/hashed_crossing.py

```diff
@@ -105,42 +107,64 @@ def __init__(
         self.supports_jit = False
 
     def compute_output_shape(self, input_shape):
-        if (
-            not len(input_shape) == 2
-            or not isinstance(input_shape[0], tuple)
-            or not isinstance(input_shape[1], tuple)
-        ):
-            raise ValueError(
-                "Expected as input a list/tuple of 2 tensors. "
-                f"Received input_shape={input_shape}"
-            )
-        if input_shape[0][-1] != input_shape[1][-1]:
-            raise ValueError(
-                "Expected the two input tensors to have identical shapes. "
-                f"Received input_shape={input_shape}"
-            )
-
-        if not input_shape:
-            if self.output_mode == "int":
-                return ()
-            return (self.num_bins,)
-        if self.output_mode == "int":
-            return input_shape[0]
-
-        if self.output_mode == "one_hot" and input_shape[0][-1] != 1:
-            output_shape = tuple(input_shape[0]) + (self.num_bins,)
-        output_shape = tuple(input_shape[0])[:-1] + (self.num_bins,)
-        return output_shape
+        self._check_at_least_two_inputs(input_shape)
+        return tf_utils.compute_shape_for_encode_categorical(
+            input_shape[0], self.output_mode, self.num_bins
+        )
 
     def call(self, inputs):
-        outputs = self.layer.call(inputs)
+        self._check_at_least_two_inputs(inputs)
+        inputs = [layer_utils.ensure_tensor(x) for x in inputs]
+        self._check_input_shape_and_type(inputs)
+
+        # Uprank to rank 2 for the cross_hashed op.
+        rank = inputs[0].shape.rank
+        if rank < 2:
+            inputs = [tf_utils.expand_dims(x, -1) for x in inputs]
+        if rank < 1:
+            inputs = [tf_utils.expand_dims(x, -1) for x in inputs]
+
+        # Perform the cross and convert to dense
+        outputs = tf.sparse.cross_hashed(inputs, self.num_bins)
+        outputs = tf.sparse.to_dense(outputs)
+
+        # Fix output shape and downrank to match input rank.
+        if rank == 2:
+            # tf.sparse.cross_hashed output shape will always be None on the
+            # last dimension. Given our input shape restrictions, we want to
+            # force shape 1 instead.
+            outputs = tf.reshape(outputs, [-1, 1])
+        elif rank == 1:
+            outputs = tf.reshape(outputs, [-1])
+        elif rank == 0:
+            outputs = tf.reshape(outputs, [])
+
+        # Encode outputs.
+        outputs = tf_utils.encode_categorical_inputs(
+            outputs,
+            output_mode=self.output_mode,
+            depth=self.num_bins,
+            sparse=self.sparse,
+            dtype=self.compute_dtype,
+        )
         if (
             backend.backend() != "tensorflow"
             and not backend_utils.in_tf_graph()
         ):
             outputs = backend.convert_to_tensor(outputs)
         return outputs
 
+    def compute_output_signature(self, input_specs):
```

Comment:
This method shouldn't be included

Reply:
Since this is a added utility for only getting signature, This dosen't affect core function for this layer.